src/bitops.c

```diff
@@ -457,22 +457,23 @@ int getBitfieldTypeFromArgument(client *c, robj *o, int *sign, int *bits) {
  * so that the 'maxbit' bit can be addressed. The object is finally
  * returned. Otherwise if the key holds a wrong type NULL is returned and
  * an error is sent to the client. */
-robj *lookupStringForBitCommand(client *c, uint64_t maxbit, int *dirty) {
+robj *lookupStringForBitCommand(client *c, uint64_t maxbit, size_t *stringGrowSize) {
+    int _stringGrowSize = 0;
     size_t byte = maxbit >> 3;
     robj *o = lookupKeyWrite(c->db,c->argv[1]);
     if (checkType(c,o,OBJ_STRING)) return NULL;
-    if (dirty) *dirty = 0;
 
     if (o == NULL) {
         o = createObject(OBJ_STRING,sdsnewlen(NULL, byte+1));
         dbAdd(c->db,c->argv[1],o);
-        if (dirty) *dirty = 1;
+        _stringGrowSize = byte + 1;
```

Comment:
we can save a variable and avoid to assign it if it's NULL; 
```suggestion
        if (stringGrowSize) *stringGrowSize = byte + 1;
```

Reply:
Tried to avoid nested conditions and opt branching. A matter of style. Lazily prefer to leave it as is, unless you have strong opinion about it.

Reply:
at least we don't need to assign `byte + 1` or `sdslen(o->ptr) - oldlen` to `_stringGrowSize` if _stringGrowSize is NULL.